### ISSUE TITLE
Peer validation by Total Terminal Difficulty (TTD)

### ISSUE THREAD

**Issue by fab-10** [hyperledger/besu#3862]
### Description

The switch to PoS happens when the PoW chain total difficulty reaches the value speficied by TTD, after that event, the difficulty will never increase again, and so total difficulty will remain fixed at the last value.
It could happen that not all the Execution Clients will agree with the switch to PoS, and they could continue to follow the PoW chain, causing forks that will continue to increase the total difficulty, with values that could be higher than TTD. These peers are not useful for Besu after it switched to PoS, and viceversa, so it make sense to extend the peer validation mechanism to avoid that Besu estabilishes a connection with them.

Peer validation rules are defined in the package `org.hyperledger.besu.ethereum.eth.peervalidation`.

Basically the new `TotalTerminalDifficultyPeerValidator`, should be enabled only if the TTD value is present in the genesis file, and should behave like that, if TTD has been reached (i.e. Besu switched to PoS), then a peer is invalid if its total difficulty is not equal to our total difficulty.

This also helps with the peering on shadow forks testnets.

### Acceptance Criteria
* Before the switch to PoS nothing changes
* After the switch to PoS clients that continue to follow PoW, will be considered invalid on connection

**Comment by non-fungible-nelson** [comment]
Closed as irrelevant post-merge
